### ISSUE TITLE
Cannot run ASR_with_NeMo.ipynb example

### ISSUE THREAD

**Issue by BenAWang10137** [NVIDIA/NeMo#5175]
I cannot run the example through CoLab, it stucks in loading in pre-trained model.

# This line will download pre-trained QuartzNet15x5 model from NVIDIA's NGC cloud and instantiate it for you
quartznet = nemo_asr.models.EncDecCTCModel.from_pretrained(model_name="QuartzNet15x5Base-En")

AttributeError: module 'math' has no attribute 'prod'

If I use v1.12.0 instead main branch, there will be no problem.

**Comment by michael-so** [comment]
Hi, any update on this issue? I have also experienced this issue while creating a new model and loading existing ones.

**Comment by titu1994** [comment]
Can you post the full stack trace ?

**Comment by nabil6391** [comment]
I am facing the same issue while trying to finetune and load a model. 

The issue seems to be in asr_confidence_utils.py and its because google colab is running python 3.7 which does not have math.prod function.

```
AttributeError                            Traceback (most recent call last)
[<ipython-input-13-bfd089d89622>](https://localhost:8080/#) in <module>
      1 # model = nemo_asr.models.ASRModel.from_pretrained("stt_en_citrinet_1024", map_location='cpu')
----> 2 model = nemo_asr.models.ASRModel.from_pretrained("stt_en_conformer_ctc_large", map_location='cpu')
      3 
      4 # model = nemo_asr.models.EncDecCTCModelBPE.restore_from(saved_model+"/model", map_location='cpu')

11 frames
[/usr/local/lib/python3.7/dist-packages/nemo/core/classes/common.py](https://localhost:8080/#) in from_pretrained(cls, model_name, refresh_cache, override_config_path, map_location, strict, return_config, trainer, save_restore_connector)
    855             return_config=return_config,
    856             trainer=trainer,
--> 857             save_restore_connector=save_restore_connector,
    858         )
    859         return instance

[/usr/local/lib/python3.7/dist-packages/nemo/core/classes/modelPT.py](https://localhost:8080/#) in restore_from(cls, restore_path, override_config_path, map_location, strict, return_config, save_restore_connector, trainer)
    315         cls.update_save_restore_connector(save_restore_connector)
    316         instance = cls._save_restore_connector.restore_from(
--> 317             cls, restore_path, override_config_path, map_location, strict, return_config, trainer
    318         )
    319         if isinstance(instance, ModelPT):

[/usr/local/lib/python3.7/dist-packages/nemo/core/connectors/save_restore_connector.py](https://localhost:8080/#) in restore_from(self, calling_cls, restore_path, override_config_path, map_location, strict, return_config, trainer)
    234         # (original .nemo behavior)
    235         loaded_params = self.load_config_and_state_dict(
--> 236             calling_cls, restore_path, override_config_path, map_location, strict, return_config, trainer,
    237         )
    238         if not isinstance(loaded_params, tuple):

[/usr/local/lib/python3.7/dist-packages/nemo/core/connectors/save_restore_connector.py](https://localhost:8080/#) in load_config_and_state_dict(self, calling_cls, restore_path, override_config_path, map_location, strict, return_config, trainer)
    156                 # get the class
    157                 calling_cls._set_model_restore_state(is_being_restored=True, folder=tmpdir)
--> 158                 instance = calling_cls.from_config_dict(config=conf, trainer=trainer)
    159                 instance = instance.to(map_location)
    160                 # add load_state_dict override

[/usr/local/lib/python3.7/dist-packages/nemo/core/classes/common.py](https://localhost:8080/#) in from_config_dict(cls, config, trainer)
    503                     if prev_error:
    504                         logging.error(prev_error)
--> 505                     raise e
    506 
    507         if not hasattr(instance, '_cfg'):

[/usr/local/lib/python3.7/dist-packages/nemo/core/classes/common.py](https://localhost:8080/#) in from_config_dict(cls, config, trainer)
    495                     accepts_trainer = Serialization._inspect_signature_for_trainer(cls)
    496                     if accepts_trainer:
--> 497                         instance = cls(cfg=config, trainer=trainer)
    498                     else:
    499                         instance = cls(cfg=config)

[/usr/local/lib/python3.7/dist-packages/nemo/collections/asr/models/ctc_bpe_models.py](https://localhost:8080/#) in __init__(self, cfg, trainer)
     69             cfg.decoder["num_classes"] = len(vocabulary)
     70 
---> 71         super().__init__(cfg=cfg, trainer=trainer)
     72 
     73         # Setup decoding objects

[/usr/local/lib/python3.7/dist-packages/nemo/collections/asr/models/ctc_models.py](https://localhost:8080/#) in __init__(self, cfg, trainer)
     92                 self.cfg.decoding = decoding_cfg
     93 
---> 94         self.decoding = CTCDecoding(self.cfg.decoding, vocabulary=self.decoder.vocabulary)
     95 
     96         # Setup metric with decoding strategy

[/usr/local/lib/python3.7/dist-packages/nemo/collections/asr/metrics/wer.py](https://localhost:8080/#) in __init__(self, decoding_cfg, vocabulary)
    847         self.labels_map = dict([(i, vocabulary[i]) for i in range(len(vocabulary))])
    848 
--> 849         super().__init__(decoding_cfg=decoding_cfg, blank_id=blank_id)
    850 
    851     def _aggregate_token_confidence(self, hypothesis: Hypothesis) -> List[float]:

[/usr/local/lib/python3.7/dist-packages/nemo/collections/asr/metrics/wer.py](https://localhost:8080/#) in __init__(self, decoding_cfg, blank_id)
    202 
    203         # initialize confidence-related fields
--> 204         self._init_confidence(self.cfg.get('confidence_cfg', None))
    205 
    206         # we need timestamps to extract non-blank per-frame confidence

[/usr/local/lib/python3.7/dist-packages/nemo/collections/asr/parts/utils/asr_confidence_utils.py](https://localhost:8080/#) in _init_confidence(self, confidence_cfg)
    211 
    212         # define aggregation functions
--> 213         self.confidence_aggregation_bank = get_confidence_aggregation_bank()
    214         self._aggregate_confidence = self.confidence_aggregation_bank[self.word_confidence_aggregation]
    215 

[/usr/local/lib/python3.7/dist-packages/nemo/collections/asr/parts/utils/asr_confidence_utils.py](https://localhost:8080/#) in get_confidence_aggregation_bank()
    143         dictionary with functions.
    144     """
--> 145     return {"mean": (lambda x: sum(x) / len(x)), "min": min, "max": max, "prod": math.prod}
    146 
    147 

AttributeError: module 'math' has no attribute 'prod'
```

**Comment by hamjam** [comment]
It turns out you're using python3.7 but NeMo and specifically `math.prod` requires python3.8.

**Comment by titu1994** [comment]
This should be fixed as per #5180. Im hoping colab updates its python version soon, we require/want to support python 3.9+ now, not 3.8.
